File: C#/RPC/Hayaa.RPC.Componet/Client/RpcClient.cs

namespace Hayaa.RPC.Componet
{
    /// <summary>
    /// rpc客户端主类
    /// </summary>
    public class RpcClient
    {
        private static RPCConfig g_config = null;
        public  void Run(RPCConfig config)
        {
            g_config = config;
            ServiceFactory.initService(config.ConsumerConfiguation.Services);
            ClientHelper.Instance.Init(config.ConsumerConfiguation.Services);
        }
        internal static RPCConfig GetConfig()
        {
            return g_config;
        }
    }
}

File: Java/hayaarpcv2/hayaa.rpc.client/src/main/java/hayaa/rpc/client/RpcClient.java
package hayaa.rpc.client;


import hayaa.rpc.common.config.RpcConfig;

import java.util.List;
import java.util.stream.Collectors;

/**
 * rpc客户端主类
 */
public class RpcClient {
    private static RpcConfig g_config = null;
    /**
     * rpc客户端启动入口函数
     */
    public synchronized void run(RpcConfig config) {
        g_config = config;
        List<String> interfaceNameList =config.getConsumerConfiguation().getServices().stream().map(RpcConfig.ServiceConfig::getInterfaceName).collect(Collectors.toList());
        ServiceFactory.initService(interfaceNameList);
        ClientHelper.get_instance().init(config.getConsumerConfiguation());
    }
    public static RpcConfig GetConfig()
    {
        return g_config;
    }
}
